### DISCUSSION TITLE
JavaScript Conclusion: Add brief mention of topics not covered in the course

### DISCUSSION THREAD

**Discussion by nik-rev** [TheOdinProject/curriculum#27284] [Ideas]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [x] Would you like to work on this issue?

### Describe your suggestion

When coming into the JS course, I was expecting it to cover _everything_, but it doesn't. I understand that there's no need to cover something if there isn't a genuine need for it, for example the JS course doesn't have a lesson on Map/set or Generators as two examples. 

However, I don't think it would hurt to add a mention of topics that may be worth looking into that weren't covered in the course. Add a brief list of topics and recommended resources to learn about it + include a warning telling learners that it may be worth coming back to this at a later date and these topics aren't required to learn about for now, and a warning against rabbit holes.

But I think it'd be useful as a sort of reference that people can come back to if they desire. For example it could be something like this (very roughly):

### Concepts worth looking into at a later date

The JavaScript course doesn't cover everything there is in the language, so if you would like to learn more, here are some concepts and recommended resources for them. 

⚠️ **Important notice**
Please note that you should continue with the curriculum and not deep-dive into any of these topics for now. Use this section as a reference later on when you want to learn more about the language when you've become familiar with later TOP sections.

[Concept 1]: [Brief explanation of why this is useful]
- You may want to check out <a href="#">resource 1</a> because of x
- Also, it may be useful to look into <a href="#">resource 2</a> because of y
- This <a href="#">resource 3</a> also goes in-depth about z 

[Concept 2] ...
[Concept 3] ...
... 

Let me know what you think about this!

### Path

Ruby / Rails, Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-conclusion

**Comment by CouchofTomato** [comment]
@TheOdinProject/javascript Can someone give their opinion on this please.

**Comment by thatblindgeye** [comment]
This is essentially what our "Additional resources" sections act as. My 2 cents: adding additional resources to this lesson may not be bad, but there's always going to be a lot of stuff our courses don't cover. Deciding what's worth calling out as a "we don't cover this, but here's something to look into if you want" kind of resource can be tricky. Should we call stuff out that doesn't/may not have much if any impact within the context of the curriculum, or just stuff that could be utilized moving forward if a user wants to look into it? 

Then there's whether it will send users down a rabbit hole, causing them to put off continuing through the curriculum for something that may not pertain to the curriculum just for the sake of additional JS topics.

I wouldn't be opposed to adding additional resources here, but they'd need to be carefully chosen.

**Comment by CouchofTomato** [comment]
I've converted to a discussion for now because this seems like something that would need to be fleshed out more.

**Comment by nik-rev** [comment]
So I've done some research here and there and just throwing some ideas together to see what people think. None of these are covered by the JS course and could use a mention in additional resources

- [`Proxy` and `Reflect` API.](https://youtu.be/9jl1lhFylJs).
- [Iterators and generators](https://www.digitalocean.com/community/tutorials/understanding-generators-in-javascript).
- Mention of `??=`, `||=` and `&&=`.
- `Map`, `Set`, `WeakMap`, `WeakSet`
- [`Currying`](https://javascript.info/currying-partials)

**Comment by thatblindgeye** [comment]
While all of those are interesting topics and can be helpful, it all comes back to at what point do we *not* include something? As I mentioned, there's a ton of concepts not mentioned in the JS course, and we can't mention everything as additional content that users can look into if they want, so it kinda becomes how do we decide what's worth including vs not including. Basing it on difficulty of the concept can be pretty subjective and may not always work. 

I can't recall where, but I'm pretty sure I actually expressed an opinion of removing these sort of "Conclusion" lessons as for the most part they follow a pretty standard "congrats, here's what you learned, here's some stuff that may be coming up, here's a feedback form", most of which could be added at the end of the final project of a course. That's a whole other conversation, though.

That said, our [conclusion lesson in the Rails course](https://www.theodinproject.com/lessons/ruby-on-rails-conclusion) does sort of set a precedent of including this additional content, though that content was added 6+ years ago originally. So having input from all maintainers, not just JS maintainers, would make sense here. @TheOdinProject/maintainers

**Comment by 01zulfi** [comment]
I'm on the fence on listing "things to look at" in our conclusion lesson. On one hand, I don't there's any harm in it and would encourage learners to go out and learn more. But, these topics would be subjective to different learners and maintainers and _choosing_ what to include or not would be unnecessary overhead.

**Comment by mao-sz** [comment]
Closing as it's not something we're planning to go into yet.
